rfcs/button-component.md

```diff
@@ -1,5 +1,50 @@
 # Button Component
 
+### Usage
+
+#### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Button, KIND} from 'baseui/button';
+
+export default () => <Button kind={KIND.primary}>Submit</Button>;
+```
+
+#### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {
+  Button,
+  StyledStartEnhancer,
+  StyledEndEnhancer,
+  SIZE,
+  KIND,
+  SHAPE,
+} from 'baseui/button';
+
+export default () => (
+  <Button
+    startEnhancer={
+      <StyledStartEnhancer>
+        <SvgIcon />
+      </StyledStartEnhancer>
+    }
+    endEnhancer={
+      <StyledEndEnhancer>
+        <SvgIcon />
+      </StyledEndEnhancer>
```

Comment:
Unrelated to @gergelyke 's changes but @abmai @kahwee why do we need StyledEnhancer here? Shouldn't it automatically be using these internally already? Seems like you should only need to import `Styled*` when you want to use the `overrides` mechanism.

Reply:
Hm my guess is probably just an oversight in writing these. It should be used internally already yes. cc @kahwee